Format 2103 User Guide (#301)
AY2324S1-CS2103T-F11-1/tp: docs/UserGuide.md

* Enrols the member at the specified `MEMBER_INDEX` to the event at the specified `EVENT_INDEX` with `NUMBER_OF_HOURS` specifying the number of hours that the member contributed and `REMARK` stating extra remarks about the member and event.

Examples:
* `enrol m/1 e/1 h/2 r/Role: Photographer` enrols member at **member index** 1 in the member list to the event at **event index** 5 in the event list, where the member had 3 **hours** of contributions to that event and has a **remark** stating that the member "did planning".
* `enrol m/5 e/1` enrols the member at **member index** 5 in the member list to the event at **event index** 1 in the event list.

<hr class="command-separator">

> #### Edit an enrolment: `editEnrolment`

Reviewer: ```suggestion
#### Edit an Enrolment: `editEnrolment`
```

* Edits the specified `MEMBER_INDEX`'s enrolment of the event at the specified `EVENT_INDEX` with `NUMBER_OF_HOURS` specifying the number of hours that the member contributed and `REMARK` stating extra remarks about the member and event.

<div markdown="block" class="alert alert-info">
:information_source: Remember to provide at least one field to be edited!
</div>

Examples:

> * `editEnroment m/1 e/2 h/5 r/Role: Exco` edits the enrolment of the member at **member index** 1 to the event at **event index** 2, changing the number of **hours** to 5 and the **remark** to "Role: Exco".

Reviewer 2: ```suggestion
* `editEnrolment m/1 e/2 h/5 r/Role: Exco` edits the enrolment of the member at **member index** 1 to the event at **event index** 2, changing the number of **hours** to 5 and the **remark** to "Role: Exco".
```

* `unenrol m/5 e/1` unenrols the member at **member index** 5 in the member list from the event at **event index** 1 in the event list.

<div markdown="block" class="alert alert-info">:information_source: The `MEMBER_INDEX`/`EVENT_INDEX` parameter refers to the index number shown in the **currently displayed** member/event list.
</div>
<hr class="feature-class-separator">

<div style="page-break-after: always;"></div>

### View Commands
In CCACommander, you can quickly change the view to fit your current use case, be it to check for the enrolment of an event, 
searching for a member and more!

#### View Events of Member : `viewMember`

Lists all the events of a specified member index.

Format: `viewMember MEMBER_INDEX`

* Views the events of the member at the specified `MEMBER_INDEX`.
* `MEMBER_INDEX` refers to the index number shown in the **currently displayed** member list.

Examples:
* `viewMember 3` displays events of the 3rd member in the member list.

<figure>
    <img src="images/viewMemberPreCommand.png"
         alt="viewMember Pre Command">
    <figcaption>Fig 5: Before executing the <code>viewMember</code> command</figcaption>
</figure>
<br>
<figure>

>     <img src="images/viewMemberPostCommand.png"

Reviewer: Change the screenshot to the one without fullstop.

         alt="viewMember Post Command">
    <figcaption>Fig 6: After executing the <code>viewMember</code> command</figcaption>
</figure>

<hr class="command-separator">

#### View Members of Event : `viewEvent`

Lists all the members of a specified event index.

Format: `viewEvent EVENT_INDEX`
* Views the members of the event at the specified `EVENT_INDEX`.
* `EVENT_INDEX` refers to the index number shown in the **currently displayed** event list.

Examples:
* `viewEvent 1` displays members of the 1st event in the event list.

<figure>
    <img src="images/viewEventPreCommand.png"
         alt="viewEvent Pre Command">
    <figcaption>Fig 7: Before executing the <code>viewEvent</code> command</figcaption>
</figure>
<br>
<figure>

>     <img src="images/viewEventPostCommand.png"

Reviewer: Change the screenshot to the one without fullstop.

         alt="viewEvent Post Command">
    <figcaption>Fig 8: After executing the <code>viewEvent</code> command</figcaption>
</figure>

<hr class="command-separator">

#### Find Member in member list : `findMember`

Finds and lists member(s) whose name(s) contain the provided `MEMBER_NAME`.

Format: `findMember MEMBER_NAME [MORE_NAMES]...`
* Finds and lists member(s) whose name(s) contain the specified `MEMBER_NAME`.
* More than 1 `MEMBER_NAME` can be provided to find more members.

Examples:

> * `findMember alice` displays the member(s) whose name(s) contain 'alice'

Reviewer: ```suggestion
* `findMember alice` displays the member(s) whose name(s) contain "alice"
```

> * `findMember alice bob charlie` displays the member(s) whose name(s) contain 'alice', 'bob' and/or 'charlie'.

Reviewer: ```suggestion
* `findMember alice bob charlie` displays the member(s) whose name(s) contain "alice", "bob" and/or "charlie".
```

<div markdown="block" class="alert alert-info">:information_source:
The <code>findMember</code> command requires the names to be typed in full, meaning "Nich" will not suffice for finding "Nicholas". 
To make it convenient for you, the names do not have to be capitalised! 
</div>

<hr class="command-separator">

#### Find Event in event list : `findEvent`

Finds and lists event(s) which name(s) contain the provided `EVENT_NAME`.

Format: `findEvent EVENT_NAME [MORE_NAMES]...`
* Finds and lists event(s) which name(s) contain the specified `EVENT_NAME`.
* More than 1 `EVENT_NAME` can be provided to find more events.

Examples:

> * `findEvent party` displays the event(s) which name(s) contain 'party'

Reviewer: ```suggestion
* `findEvent party` displays the event(s) which name(s) contain "party"
```

> * `findEvent party marathon gaming` displays the event(s) which name(s) contain 'party', 'marathon' and/or 'gaming'.

Reviewer: ```suggestion
* `findEvent party marathon gaming` displays the event(s) which name(s) contain "party", "marathon" and/or "gaming".
```

<div markdown="block" class="alert alert-info">:information_source:

> The <code>findEvent</code> command requires the names to be typed in full, meaning "Nich" will not suffice for finding "Nicholas". 

Reviewer 3: Do you want to make it more in line with Event?
```suggestion
The <code>findEvent</code> command requires the names to be typed in full, meaning "Par" will not suffice for finding "Party". 
```

> The <code>findEvent</code> command requires the names to be typed in full, meaning "Nich" will not suffice for finding "Nicholas". 

Reviewer: ```suggestion
The <code>findEvent</code> command requires the names to be typed in full, meaning "Par" will not suffice for finding "Party". 
```

To make it convenient for you, the names do not have to be capitalised! 
</div>

<hr class="command-separator">

#### List all Members and all Events : `list`
List all members and all events in the CCA in two separate columns.

Format: `list`

> A GUI similar to the one below will be shown after entering the command.

Reviewer: Suggestion remove this line or to add in a screenshot for this.

* `unenrol`
* `editEnrolment`

<hr class="command-separator">

#### Redoing a command: `redo`

Redoes a command that the user has undone previously.

Format: `redo`

<hr class="command-separator">

#### Clear all Members and Events: `clear`
Clears all Member and Event entries from CCACommander.

Format: `clear`

<div markdown="block" class="alert alert-primary">:bulb: Accidentally cleared your list? 
Fret not as you can use the `undo` command to recover your data! 

</div>

<hr class="command-separator">

#### Help: `help`

Displays a pop-out window that shows a link to this User Guide.

Format: `help`

<figure>
    <img src="images/helpWindow.png"
         alt="Help Window">
    <figcaption>Fig 9: A help pop-up window will appear upon entering the <code>help</code> command</figcaption>
</figure>

<hr class="command-separator">

#### Exit: `exit`

Closes CCACommander and its display window.

Format: `exit`

<div markdown="block" class="alert alert-info">:information_source: The <code>exit</code> command exists for your convenience. 

> It performs the same task as the cross in the top bar and the Exit button in the File sub-menu item

Reviewer: ```suggestion
It performs the same task as the cross in the top bar and the Exit button in the File sub-menu item.
```

</div>

<hr class="command-separator">

#### Saving the data

CCACommander data are saved automatically as a JSON file `[JAR file location]/data/ccacommander.json`. Advanced users are welcome to update data directly by editing that data file.

<div markdown="span" class="alert alert-warning">:exclamation: **Caution:**
If your changes to the data file makes its format invalid, CCACommander will discard all data and start with an empty data file at the next run. Hence, it is recommended to take a backup of the file before editing it.
</div>

<hr class="feature-class-separator">

<div style="page-break-after: always;"></div>
## FAQ

**Q**: How do I transfer my data to another Computer?<br>

> **A**: Install the app in the other computer and overwrite the empty data file it creates with the file that contains the data of your previous CCACommander home folder.

Reviewer 2: I think home folder was never specified cause it was deleted in prev section. Should we still say so?

Once your laptop is recharged, you can regain access to all your stored information.

--------------------------------------------------------------------------------------------------------------------

## Known Issues

1. **When using multiple screens**, if you move the application to a secondary screen, and later switch to using only the primary screen, the GUI will open off-screen. The remedy is to delete the `preferences.json` file created by the application before running the application again.
2. **Edit** commands go back to the main view no matter what view you were in.

--------------------------------------------------------------------------------------------------------------------

## Glossary
This glossary provides definitions for the special terms used in this user guide.

**CCA**: Co-curricular activity.

**Command**: The text typed in by the user as a CCA Head for CCACommander.

**Index**: The number beside the corresponding entry in a list.

**Tag**: A word or phrase that labels a member or an event.

--------------------------------------------------------------------------------------------------------------------

<div style="page-break-after: always;"></div>

> ## Command Summary

Reviewer: Reorder commands to match order in the features description

  </tr>
  <tr>
    <td class="tg-0pky command"><code>findMember</code> / <code>findEvent</code></td>

>     <td class="tg-0pky">MEMBER_NAME / EVENT_NAME / MORE_NAMES</td>

Reviewer 3: Should this stay as the same for the 2103T side since our error message says KEYWORD
```suggestion
    <td class="tg-0pky">KEYWORD / MORE_KEYWORDS</td>
```

Author: I will copy over the table from 2101 side after its done
